app/jobs/reports/monthly_key_metrics_report.rb

```diff
@@ -6,47 +6,42 @@ class MonthlyKeyMetricsReport < BaseReport
 
     attr_reader :report_date
 
+    Report = Struct.new(:metadata, :table, :csv_name)
+
     def perform(date = Time.zone.today)
       @report_date = date
 
-      account_reuse_table = account_reuse_report.account_reuse_report
-      total_profiles_table = account_reuse_report.total_identities_report
-      account_deletion_rate_table = account_deletion_rate_report.account_deletion_report
-      total_user_count_table = total_user_count_report.total_user_count_report
-
-      upload_to_s3(account_reuse_table, report_name: 'account_reuse')
-      upload_to_s3(total_profiles_table, report_name: 'total_profiles')
-      upload_to_s3(account_deletion_rate_table, report_name: 'account_deletion_rate')
-      upload_to_s3(total_user_count_table, report_name: 'total_user_count')
+      reports = []
+      reports << Report.new(
+        metadata: account_reuse_report.account_reuse_report_metadata,
+        table: account_reuse_report.account_reuse_report,
+        csv_name: 'account_reuse'
+      )
+      reports << Report.new(
+        metadata: account_reuse_report.total_identities_report_metadata,
+        table: account_reuse_report.total_identities_report,
+        csv_name: 'total_profiles'
+      )
+      reports << Report.new(
+        metadata: account_deletion_rate_report.account_deletion_report_metadata,
+        table: account_deletion_rate_report.account_deletion_report,
+        csv_name: 'account_deletion_rate'
+      )
+      reports << Report.new(
+        metadata: total_user_count_report.total_user_count_report_metadata,
+        table: total_user_count_report.total_user_count_report,
+        csv_name: 'total_user_count'
+      )
```

Comment:
Instead of defining tables (12-15), and then uploading them to s3 (17-20), and then putting them into `email_tables` (22-49), we build an array of `Report` structs that contain the table, and also the metadata that was previously in hashes.

Reply:
I like it! We could probably inline all the `metadata` bits too?

Reply:
I don't follow what you mean about inlining `metadata` here; can you say more?

Reply:
inlining the metadata meaning just have the hash directly here:
```ruby
Report.new(
  table: total_user_count_report.total_user_count_report,
  csv_name: 'total_user_count',
  metadata: { title: ''Total user count (all-time)' },
)
```